src/npc/update/update_npc.c

```diff
@@ -0,0 +1,93 @@
+/*
+** EPITECH PROJECT, 2022
+** myrpg
+** File description:
+** updat npc
+*/
+
+#include <SFML/Graphics/Rect.h>
+#include <SFML/Graphics/Sprite.h>
+#include <SFML/Window/Keyboard.h>
+#include <stdbool.h>
+#include <SFML/System/Vector2.h>
+#include <SFML/Window/Export.h>
+#include "my_bgs.h"
+#include "my_dico.h"
+#include "my_json.h"
+#include "my_macro.h"
+#include "my_puts.h"
+#include "my_rpg.h"
+#include "npc.h"
+
+static const sfKeyCode key_interract = sfKeyA;
+
+static float refresh_rate = 1.0 / 10;
+
+static void call_callback_npc(object_t *npc, scene_t *scene, window_t *win)
+{
+    npc_data_func_t *callback = NULL;
+
+    callback = dico_t_get_value(npc->components, npc_data_callback);
+    if (callback == NULL) {
+        return;
+    }
+    callback->callback(npc, scene, win);
+}
+
+static bool check_collid_player(object_t *npc, player_t *player,
+    scene_t *scene, window_t *win)
+{
+    sfFloatRect pos[3] = {0};
+
+    if (player == NULL || player->obj == NULL) {
+        return (false);
+    }
+    pos[0] = sfSprite_getGlobalBounds(player->obj->drawable.sprite);
+    pos[1] = sfSprite_getGlobalBounds(npc->drawable.sprite);
+    if (sfFloatRect_intersects(pos, pos + 1, pos + 2) &&
+            sfKeyboard_isKeyPressed(key_interract) == sfTrue) {
+        call_callback_npc(npc, scene, win);
+        return (true);
+    }
+    return (false);
+}
+
+static bool update_rect(object_t *npc, any_t *npcjson, float dtime)
+{
+    any_t *current = NULL;
+    any_t *rectjson = NULL;
+    any_t *time = NULL;
+
+    time = get_from_any(npcjson, "d", "time");
+    current = get_from_any(npcjson, "d", "current");
```

Comment:
current unchecked

Reply:
i check it, follow it under function that use it 👁️